processors/pubmed/processor.py

```diff
@@ -4,13 +4,67 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
+import json
+import logging
+from sqlalchemy import exc as sqlalchemy_exceptions
 from .. import base
 from . import extractors as extractors_module
+logger = logging.getLogger(__name__)
 
 
-# Module API
-
 def process(conf, conn):
     extractors = base.helpers.get_variables(
-        extractors_module, lambda x: x.startswith('extract_'))
-    base.processors.process_publications(conn, 'pubmed', extractors)
+        extractors_module, lambda x: x.startswith('extract_')
+    )
+    table_name = 'pubmed'
+
+    _process_unregistered_trials(conn, table_name, extractors)
+    base.processors.process_publications(conn, table_name, extractors)
+
+
+def _process_unregistered_trials(conn, table_name, extractors):
+    source = extractors['extract_source'](None)
+    source_id = base.writers.write_source(conn, source)
+
+    for record in base.helpers.iter_rows(conn, 'warehouse', table_name, orderby='meta_id'):
+        publication = extractors['extract_publication'](record)
+        unregistered_trial = extractors['extract_trial'](record)
+        conn['database'].begin()
+        try:
+
+            # Create/Update unregistered trial from the publication
+            if not publication['registry_ids']:
+                trial_id, is_primary = base.writers.write_trial(
+                    conn, unregistered_trial, source_id, record['meta_id']
+                )
+                base.writers.write_record(
+                    conn, record, source_id, trial_id, unregistered_trial, is_primary
+                )
+
+            # Publication is not/no longer an unregistered trial but a results document
+            else:
+                delete_record = "DELETE FROM records WHERE identifiers @> '%s'"
+                delete_record = delete_record % json.dumps(unregistered_trial['identifiers'])
+                conn['database'].query(delete_record)
+                for identifiers in publication['registry_ids']:
+                    trial = base.helpers.find_trial_by_identifiers(
+                        conn, identifiers=identifiers
+                    )
+                    if trial:
```

Comment:
If a publication has `registry_ids` but the `trial` is not found the results `document` will not be created. 
However, it might be that the `publication` *did not* have `registry_ids` before which means we have an unregistered trial for it. So first [we delete the `record`](https://github.com/opentrials/processors/blob/525b68c2f5ea7a1f8846e3ed05814a7cdd9ed801/processors/pubmed/processor.py#L46-L48) of the unregistered trial which will cause the unregistered trial to be removed by the `trial_remover` processor.